### PR TITLE
Add HHsuite recipe

### PR DESCRIPTION
**Pull request by martin-steinegger**
:information_source:
Bioconda has finished the [GCC7 migration](https://github.com/bioconda/bioconda-recipes/issues/13578). If you are dealing with C/C++ or Python package it can be that you need to rebuild other dependent packages. [Bioconda utils - update-pinning](https://bioconda.github.io/updating.html#updating-recipes-for-a-pinning-change) will assist you with that. If you have any questions please use [issue 13578](https://github.com/bioconda/bioconda-recipes/issues/13578).

----------------

* [ x ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ x ] This PR adds a new recipe.
* [ x ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).

### CODE REVIEW

File: recipes/hhsuite/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake
+    - make
+  host:
+    - openmp
+    - perl
```
Comment (bgruening) [inline recipes/hhsuite/meta.yaml]: Why is perl and python needed here? Is this also a run time dependency?

File: recipes/hhsuite/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{% set version = "v3.1.0" %}
+{% set sha256 = "e67527c8dd175f5730ca877aa506bc7aa4178cb0154d5b1a2794e1a484956c83" %}
+
+package:
+  name: hhsuite
+  version: {{ version }}
+
+build:
+  number: 1
```
Comment (bgruening) [inline recipes/hhsuite/meta.yaml]: 0

File: recipes/hhsuite/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+test:
+  commands:
+    - hhblits -h > /dev/null
+    - hhsearch -h > /dev/null
+    - hhalign -h > /dev/null
+
+about:
+  home: https://github.com/soedinglab/hh-suite
+  license: GPLv3
```
Comment (bgruening) [inline recipes/hhsuite/meta.yaml]: please add `license_file: LICENSE`

File: recipes/hhsuite/build.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+
+mkdir build
+cd build
+cmake -DHAVE_MPI=0 -DCMAKE_INSTALL_PREFIX="${PREFIX}" -DHAVE_SSE2=1 ..
+make -j 2
+make install
```
Comment (bgruening) [inline recipes/hhsuite/build.sh]: is there something like a `make test` command as well?

Comment (martin-steinegger) [reply recipes/hhsuite/build.sh]: Unfortunately, we do not have any tests for hhsuite. So there is no `make test` command.

File: recipes/hhsuite/meta.yaml (hunk 1 above)
Comment (martin-steinegger) [reply recipes/hhsuite/meta.yaml]: We have scripts in hhsuite that are written in python and perl. https://github.com/soedinglab/hh-suite/tree/master/scripts

Comment (bgruening) [reply recipes/hhsuite/meta.yaml]: ok, then this should be added as run-time deps as well.

File: recipes/hhsuite/build.sh (hunk 4 above)
Comment (bgruening) [reply recipes/hhsuite/build.sh]: Ok, let's see if this works now.

File: recipes/hhsuite/meta.yaml (hunk 3 above)
Comment (martin-steinegger) [reply recipes/hhsuite/meta.yaml]: I have troubles to edit this recipe. I do not have the copy of the repository I used to create this fomular anymore. 

If I check out a new bioconda and switch to the hhsuite branch then the recipe folder (recipes/hhsuite) is empty. Also if I try to push some new file then the following error occurs "remote: `Permission to bioconda/bioconda-recipes.git denied to martin-steinegger`.

Also, checking out PR leads to an error 
    
     git fetch origin  pull/13796/head:hhsuite

     remote: Enumerating objects: 8, done.
     remote: Counting objects: 100% (8/8), done.
     remote: Compressing objects: 100% (8/8), done.
     remote: Total 8 (delta 2), reused 0 (delta 0), pack-reused 0
     Unpacking objects: 100% (8/8), done.
     From https://github.com/bioconda/bioconda-recipes
     ! [rejected]            refs/pull/13796/head -> hhsuite  (non-fast-forward)

Do you have any idea how I can edit this recipe again? Thank you for your help!

File: recipes/hhsuite/build.sh (hunk 4 above)
Comment (martin-steinegger) [reply recipes/hhsuite/build.sh]: Thanks a lot!

**Comment by bgruening** [comment]
@martin-steinegger it failed on OSX, see https://circleci.com/gh/bioconda/bioconda-recipes/49325

It's fine on Linux. So you can skip OSX if you want or we need to figure out what's going wrong here. Unfortunately, I do not have OSX around.

**Comment by martin-steinegger** [comment]
@bgruening thanks a lot for all your help! I will try to reproduce the error. I normally compile HH-suite with the GNU compiler at OSX. Is it possible to enforce GNU? 
Also I can not push to bioconda anymore `remote: Permission to bioconda/bioconda-recipes.git denied to martin-steinegger.`. Do you know whom I can contact for permission?

**Pull request merged**